### ISSUE TITLE
Vaccinations without a "vaccination date" should also be marked as relevant and included in the status calculation [2]

### ISSUE THREAD

**Issue by cmeyerhzi** [hzi-braunschweig/SORMAS-Project#9265]
### Problem Description

Vaccinations without a vaccination date should be classified as relevant. If the vaccination date is not known, a vaccination without a vaccination date is entered. This leads to falsification of vaccination data in a later report.

### Proposed Change

vaccinations without a date should also be considered as relevant in the future.

therefore we introduce a new rule:

- [ ] if `vaccinationreportDate` is within 14 days from `onsetDate`,( if not given `reportDate`), the vaccination should be relevant

and update an existing one:

- [ ] the calculation of the  `vaccinationstatus` should be updated so that the new rule is involved

### Implementation Details

`VaccinationService.isVaccinationRelevant` methods need to be changed.

### Additional Information

Corresponding SurvNet Converter Ticket: https://github.com/hzi-braunschweig/sormas-survnet-interface/issues/316

**Comment by FredrikSchaeferVitagroup** [comment]
> if vaccinationreportDate is within 14 days from onsetDate,( if not given reportDate), the vaccination should be relevant

@cmeyerhzi This rule is not clear to me. When you say ` within 14 days from onsetDate,( if not given reportDate)`, you mean the symptom onset date of a case? Then this rule does not make sense, because this date is not mandatory, but the report date is.

Also, the relevance of vaccinations is not just calculated for cases, but also for contacts. Even though contacts are not relevant for SurvNet, it is necessary to keep the relevance calculation consistent. Other countries may be using the immunization feature, too.

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup @cmeyerhzi 

> This rule is not clear to me. When you say within 14 days from onsetDate,( if not given reportDate), you mean the symptom onset date of a case? Then this rule does not make sense, because this date is not mandatory, but the report date is.

> Also, the relevance of vaccinations is not just calculated for cases, but also for contacts. Even though contacts are not relevant for SurvNet, it is necessary to keep the relevance calculation consistent. Other countries may be using the immunization feature, too.

Taking contacts and event participants into account is definitely correct and has probably just been overlooked. It's also independent of the country and relevant for any users.

To clarify the rest:
- Our pre-existing situation in the desrcibed scenario is that we have at least one vaccination that does not have a `vaccinationDate` set.
- For those vaccinations, we want to consider them relevant if they fall in the timeframe of 14 days, considering their `reportDate` in relation to the most relevant dates of the respective entity. Those are the following:
    - cases: `onsetDate`, if not given `reportDate`
    - contacts: `lastContactDate`, if not given `reportDate`
    - event participants: event `startDate`, if not given event `endDate` (if before report date), if not given event `reportDate`
(cf. https://github.com/hzi-braunschweig/SORMAS-Project/issues/7105)

- In each of the scenarios, we fall back onto the entity `reportDate` because tis one is always mandatory

**Comment by FredrikSchaeferVitagroup** [comment]
@kwa20 Thank you for the details. There's one semantic point we don't yet understand. Why shall the vaccination be relevant when the `Vaccination.reportDate` lies WITHIN a timeframe of 14 days? And what exactly does that mean? 
For example for cases: What does it mean that the `Vaccination.reportDate` shall fall in the timeframe of 14 days, considerig the `onsetDate/Case.reportDate`? Shall the `Vaccination.reportDate` be not more that 14 before the `onsetDate/Case.reportDate`? And earlier reported vaccination shall not be relevant?

**Comment by markusmann-vg** [comment]
https://github.com/hzi-braunschweig/SORMAS-Project/issues/9415 needs to be adjusted afterwards

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup @markusmann-vg When a vaccination does not have a `vaccinationDate`, it is difficult to say whether the vaccination is relevant for the respective entity or not. Thus far, we excluded them from being relevant alltogether, however, after receiving feedback from users and the RKI, this seems to produce inconsitent numbers of vaccinations where the `vaccinationDate` is simply unknown.
Unfortunately, considering all vaccinations without a `vaccinationDate` as relevant is also somewhat problematic, because vaccinations that are being added a long time after the relevant cases dates have a low likelihood of being relevant for the case. Since the usual time for a case to be worked on is 14 days after its creation, we would like to consider such vaccinations without a known `vaccinationDate`, which fall in the timeframe of the usual management duration since relevant vaccination data will likely be collected during that time.
In other words, we would like to consider vaccinations without a `vaccinationDate` as relevant if their reportDate is before or up until 14 days after the relevant case date (`onsetDate` or `reportDate`).

Hope that helps! Please let me know if I should clarify further.

**Comment by FredrikSchaeferVitagroup** [comment]
@kwa20 Thank you for the clarification. I assume it is okay to use the end of day from the vaccination(report) date for comparison? It was so far done like that with the vaccination date, and would be most consistent.

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup Yes, that should be fine

**Comment by FredrikSchaeferVitagroup** [comment]
Should be testable on any test system, as well as locally.

**Comment by abrudanancuta** [comment]
Verified on local environment with the version: 1.75.0-SNAPSHOT (7e4e9eb)
